### PR TITLE
Support azure redis cache

### LINKED ISSUE #386: Azure Cache for Redis - PQ3
Resource Support for Azure Cache for Redis
1.Dev
2.Test
3.Documentation

### PR DESCRIPTION
**Pull request by sathish-progress**
### Description

support azure redis cache(s)

### Issues Resolved
#386 

### Check List

- [X] New functionality includes tests
- [X] All tests pass
- [X] `rake lint` passes
- [X] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: terraform/azure.tf [hunk 1]
```diff
@@ -1190,6 +1190,20 @@ resource "azurerm_storage_account" "web_app_function_db" {
...
+  capacity            = 2
+  family              = "C"
+  sku_name            = "Standard"
+  enable_non_ssl_port = false
+  minimum_tls_version = "1.2"
+
+  redis_configuration {
+    maxfragmentationmemory_reserved = "50"
+  }
+}
```
Comment (soumyo13) [inline terraform/azure.tf:1206]: Can we add a new line after ending the resource end?

File: test/integration/verify/controls/azure_redis_cache.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+resource_group_name = attribute(:resource_group, default: nil)
+inspec_redis_cache_name = attribute(:inspec_redis_cache_name, default: nil)
+
+control 'azure_redis_cache' do
+  describe azure_redis_cache(resource_group: resource_group_name, name: inspec_redis_cache_name) do
```
Comment (soumyo13) [inline test/integration/verify/controls/azure_redis_cache.rb:5]: After the control block, we should add the impact and the description. This is how we do in was. Do we have to do the same?

File: test/integration/verify/controls/azure_redis_caches.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+resource_group_name = attribute(:resource_group, default: nil)
+
+control 'azure redis caches test' do
+  describe azure_redis_caches(resource_group: resource_group_name) do
```
Comment (soumyo13) [inline test/integration/verify/controls/azure_redis_caches.rb:4]: After the control block, we should add the impact and the description. This is how we do in was. Do we have to do the same?

File: test/integration/verify/controls/azure_redis_caches.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    it { should exist }
+    its('sku_names') { should include 'Standard' }
+  end
+end
+
+control 'azure redis caches filters' do
+  describe azure_redis_caches(resource_group: resource_group_name).where(enable_non_ssl_port: true) do
```
Comment (soumyo13) [inline test/integration/verify/controls/azure_redis_caches.rb:11]: After the control block, we should add the impact and the description. This is how we do in was. Do we have to do the same?

**Review by soumyo13 — APPROVED** [review]

**Comment by soumyo13** [comment]
Left out some comments. Please check once. LGTM.

File: test/integration/verify/controls/azure_redis_cache.rb (hunk 2 above)
Comment (sathish-progress) [reply test/integration/verify/controls/azure_redis_cache.rb:5]: it is not mandatory..

File: test/integration/verify/controls/azure_redis_caches.rb (hunk 3 above)
Comment (sathish-progress) [reply test/integration/verify/controls/azure_redis_caches.rb:4]: it is not mandatory..

File: test/integration/verify/controls/azure_redis_caches.rb (hunk 4 above)
Comment (sathish-progress) [reply test/integration/verify/controls/azure_redis_caches.rb:11]: it is not mandatory..

File: docs/resources/azure_redis_cache.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+end
+
+# Redis Caches that aren't found will not exist
+describe azure_redis_cache(resource_group: 'MyResourceGroup', name: 'inspec-compliance-redis-cache') do
+  it { should_not exist }
+end
+```
+
+### be_enabled_non_ssl_port
+# Redis Cache that supports non SSL ports
```
Comment (IanMadd) [inline docs/resources/azure_redis_cache.md:108]:
```suggestion

Ensure that the Redis Cache supports non-SSL ports.

```

File: docs/resources/azure_redis_cache.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+## Properties
+
+| Property                      | Description                                                      |
+|-------------------------------|------------------------------------------------------------------|
+| id                            | Resource ID.                                                     |
+| name                          | Redis Cache Name.                                       |
+| location                      | Redis Cache Location.                                   |
+| type                          | Resource type.                                                   |
+| tags                          | Resource tags.                                                   |
+| properties.sku.name           | The unique name of the SKU, such as 'P3'.                        |
```
Comment (dkumaras) [inline docs/resources/azure_redis_cache.md:64]: The values in the tables that start with 'properties' is not inclueded in the URI Parameters tables of this link: https://docs.microsoft.com/en-us/rest/api/redis/redis/get

Comment (sathish-progress) [reply docs/resources/azure_redis_cache.md:64]: this table refers to the [properties](https://docs.microsoft.com/en-us/rest/api/redis/redis/get#redisresource) section in the doc and not the URI parameters section

File: docs/resources/azure_redis_cache.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+## Examples
+
+### Test <>
+```ruby
+describe azure_redis_cache(resource_group: 'MyResourceGroup', name: 'inspec-compliance-redis-cache') do
+  its('properties.provisioningState') { should eq 'Succeeded' }
+end
+```
+### Test <>
+```ruby
```
Comment (dkumaras) [inline docs/resources/azure_redis_cache.md:85]: Please check whether its correct: { should 'Standard' } ?

File: docs/resources/azure_redis_caches.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+## Properties
+
+|Property            | Description                                        | Filter Criteria<superscript>*</superscript> |
+|--------------------|----------------------------------------------------|-----------------|
+| ids                | A list of the unique resource ids.                 | `id`            |
+| names              | A list of name for all the Resource names.         | `name`          |
+| types              | A list of types for all the resources.             | `type`          |
+| locations          | A list of locations for all the resources.         | `location`      |
+| properties         | A list of Properties all the resources.            | `properties`    |
```
Comment (IanMadd) [inline docs/resources/azure_redis_caches.md:58]: Are the filter criteria correct? I think properties should be `property`, `tags` should be `tag`,  `sku_capacitie` should be `sku_capacity`, there's a bunch of others that are plural but should be singular.

File: docs/resources/azure_redis_cache.md (hunk 6 above)
Comment (IanMadd) [reply docs/resources/azure_redis_cache.md:64]: It is the properties that aren't matching. For example, SkuName: "The type of Redis cache to deploy. Valid values: (Basic, Standard, Premium)" I think it's just the Sku properties that don't match though.

File: docs/resources/azure_redis_cache.md (hunk 7 above)
Comment (sathish-progress) [reply docs/resources/azure_redis_cache.md:85]: done

File: docs/resources/azure_redis_cache.md (hunk 6 above)
Comment (sathish-progress) [reply docs/resources/azure_redis_cache.md]: yeah, you are right.. my bad I just updated it ..

File: docs/resources/azure_redis_caches.md (hunk 8 above)
Comment (sathish-progress) [reply docs/resources/azure_redis_caches.md:58]: Yeah got it ..

Comment (sathish-progress) [reply docs/resources/azure_redis_caches.md:58]: Yeah got it ..

Comment (sathish-progress) [reply docs/resources/azure_redis_caches.md:58]: so whatever is in fields [here](https://github.com/inspec/inspec-azure/pull/394/files#diff-dea98b49464579e78814f2989f3ab91799d558609ced436bbe1c6117b4de7ecfR27-R52) will represent the filter criteria in docs and the columns as property in docs .. 

But I did notice there were some discrepancies in the docs for the filter criteria and fixed it .. thanks for pointing out ..

File: docs/resources/azure_redis_cache.md [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+Any attribute in the response may be accessed with the key names separated by dots (`.`).
+
+## Examples
+
+### Test <>
+```ruby
+describe azure_redis_cache(resource_group: 'MyResourceGroup', name: 'inspec-compliance-redis-cache') do
+  its('properties.provisioningState') { should eq 'Succeeded' }
+end
+```
```
Comment (dkumaras) [inline docs/resources/azure_redis_cache.md:83]: Is the value should 'standard' is correct or it needs to be 'Standard_1vCores'?

File: docs/resources/azure_redis_caches.md [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+end
+```
+## Parameters
+
+The parameter should be provided for valid query
+- `resource_group`
+
+| Name                            | Description                                                                      |
+|---------------------------------|----------------------------------------------------------------------------------|
+| resource_group                  | Azure resource group that the targeted resource resides in. `MyResourceGroup`    |
```
Comment (dkumaras) [inline docs/resources/azure_redis_caches.md:48]: Ian - where to find the relevant microsoft doc for this table?

File: docs/resources/azure_redis_cache.md (hunk 9 above)
Comment (sathish-progress) [reply docs/resources/azure_redis_cache.md:83]: Standard is the right one as per doc

File: docs/resources/azure_redis_caches.md (hunk 10 above)
Comment (sathish-progress) [reply docs/resources/azure_redis_caches.md:48]: these parameters are defined for an inspec resource .. and you can't find a relevant doc at microsoft..

File: docs/resources/azure_redis_cache.md [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+### Test <>
+```ruby
+describe azure_redis_cache(resource_group: 'MyResourceGroup', name: 'inspec-compliance-redis-cache') do
+  its('properties.sku.name') { should 'Standard' }
```
Comment (IanMadd) [inline docs/resources/azure_redis_cache.md:87]:
```suggestion
  its('properties.sku.name') { should eq 'Standard' }
```

**Comment by SathishAchilles** [comment]
@dkumaras Could you please remove the file `.DS_Store` you have committed here?

**Comment by sathish-progress** [comment]
@dkumaras Could you please delete the file `.DS_Store` committed here ?

**Review by IanMadd — APPROVED** [review]

**Comment by sathish-progress** [comment]
@dkumaras  Could you please remove this one as well https://github.com/inspec/inspec-azure/blob/182eced77632f02585f37516904bc407790ab5be/.DS_Store ?

**Pull request merged**